core/src/main/java/com/netease/arctic/catalog/BaseArcticCatalog.java

```diff
@@ -659,16 +659,21 @@ protected void fillTableProperties(TableMeta meta) {
 
     protected String getDatabaseLocation() {
       if (catalogMeta.getCatalogProperties() != null) {
-        String catalogWarehouseDir = catalogMeta.getCatalogProperties().getOrDefault(
-            CatalogMetaProperties.KEY_WAREHOUSE_DIR,
-            null
-        );
-        if (!Objects.equals("/", catalogWarehouseDir) && catalogWarehouseDir.endsWith("/")) {
-          catalogWarehouseDir = catalogWarehouseDir.substring(
-              0, catalogWarehouseDir.length() - 1);
+        String catalogWarehouse = catalogMeta.getCatalogProperties().getOrDefault(
+            CatalogMetaProperties.KEY_WAREHOUSE,null);
+        if (catalogWarehouse == null) {
+          catalogWarehouse = catalogMeta.getCatalogProperties().getOrDefault(
+              CatalogMetaProperties.KEY_WAREHOUSE_DIR,null);
         }
-        if (StringUtils.isNotBlank(catalogWarehouseDir)) {
-          return catalogWarehouseDir + '/' + identifier.getDatabase();
+        if (catalogWarehouse == null) {
+          throw new NullPointerException("Catalog warehouse is null.");
```

Comment:
IllegaStateExecption maybe better ?

Reply:
I prefer to use NullPointerException when the required parameter is null.